typescript/utils/src/objects.ts

```diff
@@ -98,34 +98,87 @@ export function pick<K extends string, V = any>(obj: Record<K, V>, keys: K[]) {
   return ret as Record<K, V>;
 }
 
-// Recursively merges b into a
-// Where there are conflicts, b takes priority over a
-export function objMerge(
+/**
+ *  Returns a new object that recursively merges b into a
+ *  Where there are conflicts, b takes priority over a
+ * @param a - The first object
+ * @param b - The second object
+ * @param max_depth - The maximum depth to recurse
+ * @param mergeArrays - If true, arrays will be concatenated instead of replaced
+ */
+export function objMerge<T>(
```

Comment:
Two changes here:
1. Support setting return value type via generic (e.g. `objMerge<MyReturnValue>` instead of `objMerge() as MyReturnValue)`
2. Support merging nested arrays instead just replacing them